builtins/amp-ad.js

```diff
@@ -108,8 +108,8 @@ export function installAd(win) {
        */
       this.unlistenViewportChanges_ = null;
 
-      /** @private {boolean} */
-      this.shouldSendIntersectionChanges_ = false;
+      /** @private {IntersectionObserver} */
+      this.intersectionObserver_ = null;
```

Comment:
Initialize here? Then you can assume it always exists and delete the `#sendAdIntersection_` method